ls1intum/Artemis
# `General`: Include more information in course archive

### Checklist
#### General


#### Server
### Motivation and Context


Course archives do not contain exercise details or the problem statement for all exercise types. This PR adds this information to course archives.


### Description
Refactored the `SubmissionExportService`. Fixed the exports of File Upload submissions. Introduced a new ExerciseWithSubmissionsExportService that exports a JSON file with information about the exercise, a markdown file with the problem statement and all submissions or student repos for an exercise.  For programming exercise, we also export template, solution, test, and aux repositories.

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Navigate to Course Administration
3. Select a course that has an end date in the past or set the end date of a course to a value in the past.
4. Click on archive
5. Once finished, download the archive and verify its content.
6. Every exercise should have its own directory with the problem statement as markdown file, the exercise details json file and all submisisons.
7. Also test the export of submissions (for text, modeling, and file upload) available on the scores page of an exercise (empty submissions are always excluded)
8. Also test that the export of a single programming exercise (using the download exercise button) still works.
9. Run a plagiarism check for a programming exercise and check it still works (because it's affected by the changes in FileService)


### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage







#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [CourseExamExportService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5420/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.export/CourseExamExportService.html) | 90% | ✅|
| [ExerciseWithSubmissionsExportService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5420/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.export/ExerciseWithSubmissionsExportService.html) | 91% | ✅|
| [FileUploadExerciseWithSubmissionsExportService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5420/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.export/FileUploadExerciseWithSubmissionsExportService.html) | 100% | ✅ |
| [ModelingExerciseWithSubmissionsExportService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5420/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.export/ModelingExerciseWithSubmissionsExportService.html) | 100% | ✅ |
| [TextExerciseWithSubmissionsExportService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5420/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.export/TextExerciseWithSubmissionsExportService.html) | 100% | ✅|
| [ProgrammingExerciseExportService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5420/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.export/ProgrammingExerciseExportService.html) | 85% | ✅|

### Screenshots
